### PR TITLE
magento/graphql-ce#41: [Query] My Account > My Orders

### PR DESCRIPTION
**Pull request by sheepfy**
### IMPORTANT
This interface **Magento\CustomerGraphQl\Model\Customer\CheckCustomerAccountInterface;**
will be merged in the pull request https://github.com/magento/graphql-ce/pull/213

### Description (*)
New module to extract a list of orders by customer

### Fixed Issues (if relevant)
magento/graphql-ce#41: [Query] My Account > My Orders

### Manual testing scenarios (*)
step 1: Install chrome extension [ChromeiQL](https://chrome.google.com/webstore/detail/chromeiql/fkkiamalmpiidkljmicmjfbieiclmeij)
step 2: Set endpoint http://<your_host>/graphql?XDEBUG_SESSION_START=idekey
step 3: run grahpql query
{
  customerOrders {
    items {
      id
      increment_id
      created_at
      state
      status
      grant_total
    }
  }
}

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/graphql-ce?pullRequest=212) <br/>All committers have signed the CLA.

File: app/code/Magento/SalesGraphQl/Model/Resolver/Orders.php
Comment (paliarush) [inline app/code/Magento/SalesGraphQl/Model/Resolver/Orders.php]: For consistency purposes it is better to get info about current user from the `$context` object, it is done this way in all other resolvers.

File: app/code/Magento/SalesGraphQl/README.md
Comment (paliarush) [inline app/code/Magento/SalesGraphQl/README.md]: endpoints => queries
endpoints is REST terminology,
queries and mutations is used in GraphQL

File: app/code/Magento/SalesGraphQl/README.md
Comment (paliarush) [inline app/code/Magento/SalesGraphQl/README.md]: `endpoints for modifying an order` => `order mutations`

File: app/code/Magento/SalesGraphQl/etc/schema.graphqls
Comment (paliarush) [inline app/code/Magento/SalesGraphQl/etc/schema.graphqls]: `grant_total` => `grand_total`

File: app/code/Magento/SalesGraphQl/etc/schema.graphqls
Comment (paliarush) [inline app/code/Magento/SalesGraphQl/etc/schema.graphqls]: Why do we need to expose state on the storefront?

File: app/code/Magento/SalesGraphQl/etc/schema.graphqls
Comment (paliarush) [inline app/code/Magento/SalesGraphQl/etc/schema.graphqls]: `Orders` => `CustomerOrders`

File: app/code/Magento/SalesGraphQl/etc/schema.graphqls
Comment (paliarush) [inline app/code/Magento/SalesGraphQl/etc/schema.graphqls]: `Order` => `CustomerOrder`

File: dev/tests/api-functional/phpunit.xml.dist
Comment (paliarush) [inline dev/tests/api-functional/phpunit.xml.dist]: Looks like this file was committed by accident. Please remove.

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Sales/OrdersTest.php
Comment (paliarush) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Sales/OrdersTest.php]: Please correct indentation.

File: app/code/Magento/SalesGraphQl/Model/Resolver/Orders.php
Comment (paliarush) [inline app/code/Magento/SalesGraphQl/Model/Resolver/Orders.php]: These should be addressed or moved to new issues before we can merge this PR.

**Review by paliarush — CHANGES_REQUESTED** [review]
@sheepfy Thank you for the contribution, please address several comments before we can merge the PR.
Also there are some coding standard violations which will be highlighted by the static tests.

File: app/code/Magento/SalesGraphQl/etc/schema.graphqls
Comment (sheepfy) [reply app/code/Magento/SalesGraphQl/etc/schema.graphqls]: I will tend to provide a lot of info on the order, so the query can be made to extract just you need.

File: app/code/Magento/SalesGraphQl/Model/Resolver/Orders.php
Comment (sheepfy) [reply app/code/Magento/SalesGraphQl/Model/Resolver/Orders.php]: those todo's can be removed for now, i will come with updates in order to add more info on orders , if that's ok

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Sales/OrdersTest.php
Comment (sheepfy) [reply dev/tests/api-functional/testsuite/Magento/GraphQl/Sales/OrdersTest.php]: there are 4 spaces? What's wrong?

**Comment by sheepfy** [comment]
Using the $context in resolver is a bad thing. You should consider to remove the dependency of $context and inject just what you need.

Anyway, i changed for now.

**Review by paliarush — APPROVED** [review]

File: app/code/Magento/SalesGraphQl/etc/schema.graphqls
Comment (misha-kotov) [reply app/code/Magento/SalesGraphQl/etc/schema.graphqls]: State is an internal concept, on storefront we normally expose only Status. 
```suggestion
```

Comment (sheepfy) [reply app/code/Magento/SalesGraphQl/etc/schema.graphqls]: Thanks for info. I changed already

**Review by paliarush — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
@sheepfy thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: app/code/Magento/SalesGraphQl/Model/Resolver/Orders.php
Comment (paliarush) [inline app/code/Magento/SalesGraphQl/Model/Resolver/Orders.php:27]: Why concrete implementation is preferable in this case?

**Pull request merged**
